src/sentry/static/sentry/app/views/projectDetail/charts/sessionsRequest.tsx

```diff
@@ -0,0 +1,284 @@
+      return {
+        ...baseParams,
+        ...getParams(datetime),
+      };
+    }
+
+    const {period} = selection.datetime;
+    const doubledPeriod = getPeriod(
+      {period, start: undefined, end: undefined},
+      {shouldDoublePeriod: true}
+    ).statsPeriod;
+
+    return {
+      ...baseParams,
+      statsPeriod: doubledPeriod,
+    };
+  }
+
+  transformData(
+    responseData: SessionApiResponse,
+    {fetchedWithPrevious = false}
+  ): {
+    timeseriesData: Series[];
+    previousTimeseriesData: Series | null;
+    totalSessions: number;
+  } {
+    // Take the floor just in case, but data should always be divisible by 2
+    const dataMiddleIndex = Math.floor(responseData.intervals.length / 2);
+
+    // calculate the total number of sessions for this period (exclude previous if there)
+    const totalSessions = responseData.groups.reduce(
+      (acc, group) =>
+        acc +
+        group.series['sum(session)']
+          .slice(fetchedWithPrevious ? dataMiddleIndex : 0)
+          .reduce((value, groupAcc) => groupAcc + value, 0),
+      0
+    );
+
+    // TODO(project-details): refactor this to avoid duplication as we add more session charts
+    const timeseriesData = [
+      {
+        seriesName: t('This Period'),
+        data: responseData.intervals
+          .slice(fetchedWithPrevious ? dataMiddleIndex : 0)
+          .map((interval, i) => {
+            const totalIntervalSessions = responseData.groups.reduce(
+              (acc, group) =>
+                acc +
+                group.series['sum(session)'].slice(
+                  fetchedWithPrevious ? dataMiddleIndex : 0
+                )[i],
+              0
+            );
+
+            const intervalCrashedSessions =
+              responseData.groups
+                .find(group => group.by['session.status'] === 'crashed')
+                ?.series['sum(session)'].slice(fetchedWithPrevious ? dataMiddleIndex : 0)[
+                i
+              ] ?? 0;
+
+            const crashedSessionsPercent = percent(
+              intervalCrashedSessions,
+              totalIntervalSessions
+            );
+
+            return {
+              name: interval,
+              value: getCrashFreePercent(100 - crashedSessionsPercent),
+            };
+          }),
+      },
+    ];
+
+    const previousTimeseriesData = fetchedWithPrevious
+      ? {
+          seriesName: t('Previous Period'),
+          data: responseData.intervals.slice(0, dataMiddleIndex).map((_interval, i) => {
+            const totalIntervalSessions = responseData.groups.reduce(
+              (acc, group) =>
+                acc + group.series['sum(session)'].slice(0, dataMiddleIndex)[i],
+              0
+            );
+
+            const intervalCrashedSessions =
+              responseData.groups
+                .find(group => group.by['session.status'] === 'crashed')
+                ?.series['sum(session)'].slice(0, dataMiddleIndex)[i] ?? 0;
```

Comment:
These calculations are kind of a mess right now and share a lot of code with the current timeseries data. I'll think of a better abstraction as we add more session charts.